network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TapChangerRegulatingPoint.java

```diff
@@ -0,0 +1,64 @@
+package com.powsybl.network.store.iidm.impl;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.PhaseTapChanger;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.network.store.model.*;
+
+import java.util.function.Function;
+
+public final class TapChangerRegulatingPoint extends AbstractRegulatingPoint {
+    private final AbstractTapChanger tapChanger;
+
+    public TapChangerRegulatingPoint(NetworkObjectIndex index, AbstractTapChanger tapChanger,
+                                     Function<Attributes, AbstractRegulatingEquipmentAttributes> attributesGetter) {
+        super(index, attributesGetter);
+        this.tapChanger = tapChanger;
+    }
+
+    public RegulatingPointAttributes getAttributes() {
+        return attributesGetter.apply(tapChanger.getAttributes()).getRegulatingPoint();
+    }
+
+    @Override
+    public void setRegulatingTerminal(TerminalImpl<?> regulatingTerminal) {
+        TerminalImpl<?> oldRegulatingTerminal = (TerminalImpl<?>) TerminalRefUtils.getTerminal(index,
+            getAttributes().getRegulatingTerminal());
+        if (oldRegulatingTerminal != null) {
+            oldRegulatingTerminal.removeRegulatingPoint(this);
+        }
+        regulatingTerminal.setAsRegulatingPoint(this);
+        tapChanger.getTransformer().updateResource(res -> getAttributes()
+            .setRegulatingTerminal(TerminalRefUtils.getTerminalRefAttributes(regulatingTerminal)));
+        tapChanger.getTransformer().updateResource(res -> getAttributes()
+            .setRegulatedResourceType(ResourceType.convert(regulatingTerminal.getConnectable().getType())));
+    }
+
+    @Override
+    public void resetRegulationToLocalTerminal() {
+        tapChanger.getTransformer().updateResource(res -> getAttributes().setRegulatingTerminal(getAttributes().getLocalTerminal()));
+        tapChanger.getTransformer().updateResource(res -> getAttributes().setRegulatedResourceType(getAttributes().getRegulatingResourceType()));
+    }
+
+    @Override
+    public void setRegulationMode(String regulationMode) {
+        tapChanger.getTransformer().updateResource(res -> getAttributes().setRegulationMode(regulationMode));
+    }
```

Comment:
I think it's not necessary to duplicate that. We can have a method in the abstract class like getAttributes() and execute setRegulationMode and others on this like
        getIdentifiable().updateResource(res -> getAttributes().setRegulationMode(regulationMode));
Same for the InjectionRegulatingPoint
It would avoid code duplication

Reply:
done it for these method and others